### PR TITLE
Tweak: Custom links in menu items should be relative to the site address in Kit import [ED-6947]

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: tests/phpunit/elementor/core/utils/import-export/test-url.php [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+		$url = '/not-refer-to-any-post/';
+
+		// Act
+		$modified_url = Url::migrate( $url );
+
+		// Assert
+		$this->assertEquals( $url, $modified_url );
+
+	}
+}
```
Comment (StyleShit) [inline tests/phpunit/elementor/core/utils/import-export/test-url.php:167]: add empty line

File: tests/phpunit/elementor/core/utils/import-export/test-url.php [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+	/**
+	 * Plain permalink structure
+	 */
+
+	public function test_migrate__plain_to_post_name_permalink() {
+		// Arrange
+		$this->wp_rewrite->set_permalink_structure( static::PERMALINK_STRUCTURE_PLAIN );
+
+		$base_url = 'www.base.url';
```
Comment (StyleShit) [inline tests/phpunit/elementor/core/utils/import-export/test-url.php:43]: We usually use 

```suggestion
		$base_url = 'https://test.local';
```

File: tests/phpunit/elementor/core/utils/import-export/test-url.php [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		// Act
+		$modified_url = Url::migrate( $url, $base_url );
+
+		// Assert
+		$this->assertEquals( $url, $modified_url );
+	}
+
+	public function test_migrate__full_dynamic_url() {
+		// Arrange
+		$url = 'https://www.elementor.com';
```
Comment (StyleShit) [inline tests/phpunit/elementor/core/utils/import-export/test-url.php:123]: Use `https://tests.local` to be consistent

File: core/app/modules/import-export/import.php [hunk 4]
```diff
@@ -36,10 +37,11 @@ final public function run() {
 		remove_filter( 'elementor/document/save/data', [ $this, 'prevent_saving_elements_on_post_creation' ], 10 );
 
 		$map_old_new_post_ids = $this->map_old_new_post_ids( $imported_posts );
+		$original_site = isset( $manifest_data['site'] ) ? $manifest_data['site'] : null;
```
Comment (StyleShit) [inline core/app/modules/import-export/import.php:40]: maybe use `empty()` ?

File: core/app/modules/import-export/import.php [hunk 5]
```diff
@@ -93,23 +95,31 @@ private function save_elements_of_imported_posts( $map_old_new_post_ids ) {
 		}
 	}
 
-	private function update_object_id_of_imported_menu_items( $map_old_new_post_ids ) {
+	private function update_object_id_of_imported_menu_items( $map_old_new_post_ids, $original_site ) {
```
Comment (StyleShit) [inline core/app/modules/import-export/import.php:98]: add PHPDoc to describe the params (is `$original_site` a base URL?)

File: core/app/modules/import-export/import.php [hunk 6]
```diff
@@ -93,23 +95,31 @@ private function save_elements_of_imported_posts( $map_old_new_post_ids ) {
...
-			}
+			switch ( $post_meta['_menu_item_type'][0] ) {
+				case 'post_type':
+					$update_meta = update_post_meta( $new_post_id, '_menu_item_object_id', $map_old_new_post_ids[ $menu_item_object_id ] );
+
+					if ( ! $update_meta ) {
+						wp_delete_post( $new_post_id );
+					}
+					break;
+				case 'custom':
```
Comment (StyleShit) [inline core/app/modules/import-export/import.php:115]: add space

File: core/utils/import-export/url.php [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+		if ( ! empty( $base_url ) ) {
+			$url = str_replace( $base_url, '', $url );
+		}
+
+		$parsed = wp_parse_url( $url );
+
+		if ( ! empty( $parsed['path'] ) ) {
+			$page = get_page_by_path( $parsed['path'] );
+			if ( ! $page ) {
+
```
Comment (StyleShit) [inline core/utils/import-export/url.php:23]:
```suggestion

			if ( ! $page ) {
```

Comment (StyleShit) [reply core/utils/import-export/url.php:23]: add space above and remove space below

File: core/utils/import-export/url.php [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+				unset( $parse_str_query_result['page_id'] );
+
+				$parsed['query'] = http_build_query( $parse_str_query_result );
+			}
+
+			if ( empty( $permalink_query ) ) {
+				$permalink .= '?' . $parsed['query'];
+			} else {
+				$permalink .= '&' . $parsed['query'];
+			}
```
Comment (StyleShit) [inline core/utils/import-export/url.php:52]: maybe extract chunks of logic like this one to other methods, like `add_query_params()` or something (I didn't really read the whole logic to come up with a better name)

**Review by StyleShit — DISMISSED** [review]

**Review by yossi-elementor — APPROVED** [review]

**Pull request merged**